src/common/schema.cpp

```diff
@@ -6,46 +6,49 @@ namespace duckdb {
 
 // https://iceberg.apache.org/spec/#schemas
 
-// forward declaration
-static LogicalType ParseTypeValue(rest_api_objects::Type &type);
-
-static LogicalType ParseStruct(rest_api_objects::StructType &type) {
-	child_list_t<LogicalType> children;
-	for (auto &field : type.fields) {
-		// NOTE: 'id', 'required', 'doc', 'initial_default', 'write_default' are ignored for now
-		auto name = field->name;
-		auto type = ParseTypeValue(*field->type);
-		children.push_back(std::make_pair(name, type));
-	}
-	return LogicalType::STRUCT(std::move(children));
-}
-
-static LogicalType ParseList(rest_api_objects::ListType &type) {
-	// NOTE: 'element-id', 'element-required' are ignored for now
-	auto child_type = ParseTypeValue(*type.element);
-	return LogicalType::LIST(child_type);
-}
-
-static LogicalType ParseMap(rest_api_objects::MapType &type) {
-	// NOTE: 'key-id', 'value-id', 'value-required' are ignored for now
-	auto key_type = ParseTypeValue(*type.key);
-	auto value_type = ParseTypeValue(*type.value);
-	return LogicalType::MAP(key_type, value_type);
-}
-
-static LogicalType ParseTypeValue(rest_api_objects::Type &type) {
-	if (type.has_struct_type) {
-		return ParseStruct(type.struct_type);
+unique_ptr<IcebergColumnDefinition>
+IcebergColumnDefinition::ParseType(const string &name, int32_t field_id, bool required, rest_api_objects::Type &type,
+                                   optional_ptr<rest_api_objects::PrimitiveTypeValue> initial_default) {
+	auto res = make_uniq<IcebergColumnDefinition>();
+	res->id = field_id;
+	res->required = required;
+	res->name = name;
+	//! FIXME: parse the 'initial_default' into a Value
+
```

Comment:
This has changed the most, it now produces (potentially) nested `IcebergColumnDefinition`s
All the other changes kind of stem from this, previously we had `vector<IcebergColumnDefinition>`, which is now `vector<unique_ptr<IcebergColumnDefinition>`

Because of that, `IRCAPITable` can't be copied, and `IcebergSnapshot` as well. I've wrapped `IcebergSnapshot` into a `shared_ptr` because it is assigned to the new MultiFileList in `ComplexFilterPushdown`.
We likely want to cache this somewhere (the transaction?) instead, and just use `optional_ptr` in these places, but that can happen separately.